pulp_rpm/app/models.py

```diff
@@ -760,3 +768,111 @@ class ModulemdDefaults(Content):
+
+    # stage2
+    instimage = models.CharField(max_length=50)
+    mainimage = models.CharField(max_length=50)
+
+    # media
+    discnum = models.IntegerField()
+    totaldiscs = models.IntegerField()
+
+    # variant_childs = <str>[,<str>...]           ; UIDs of child variants
+    # variant_addons = <str>[,<str>...]             ; UIDs of child addons
+
+    # ; variant paths
+    # ; all paths are relative to .treeinfo location
+    packages = models.CharField(max_length=50)
+    repository = models.CharField(max_length=50)
+    source_packages = models.CharField(max_length=50)
+    source_repository = models.CharField(max_length=50)
+    debug_packages = models.CharField(max_length=50)
+    debug_repository = models.CharField(max_length=50)
+    identity = models.CharField(max_length=50)
+
+    # [general]
+    # ; WARNING.0 = This section provides compatibility with pre-productmd treeinfos.
+    # ; WARNING.1 = Read productmd documentation for details about new format.
+    # family = <str>                        ; equal to [release]/name
+    # version = <str>                       ; equal to [release]/version
+    # name = <str>                          ; equal to "$family $version"
+    # arch = <str>                          ; equal to [tree]/arch
+    # platforms = <str>[,<str>...]          ; equal to [tree]/platforms
+    # packagedir = <str>                    ; equal to [variant-*]/packages
+    # repository = <str>                    ; equal to [variant-*]/repository
+    # timestamp = <int>                     ; equal to [tree]/build_timestamp
+    # variant = <str>                       ; variant UID of first variant (sorted alphabetically)
+
+    @property
+    def checksums(self, rehash=False):
+        if getattr(self, '_sha256', None) is None or rehash is True:
+            self._sha256 = []
+            sha256 = hashlib.sha256()
+            for platform in self.tree_platforms():
+                for image in platform.images():
+                    with (image.name, "rb") as file:
+                        sha256.update(file)
+            self._sha256.append(sha256.hexdigest())
+        return self._sha256
+
+
+class TreePlatform(Model):
+    distribution_tree = models.ForeignKey(
+        DistributionTree, on_delete=models.CASCADE, related_name='tree_platforms'
+    )
+    name = models.CharField(max_length=20)
+
+
+class Images(Model):
+    platform = models.ForeignKey(TreePlatform, on_delete=models.CASCADE, related_name='images')
+    name = models.CharField(max_length=20)
+    path = models.CharField(max_length=50)
+
+
+class Addons(Model):
+    distribution_tree = models.ForeignKey(
+        DistributionTree, on_delete=models.CASCADE, related_name='addons'
+    )
+    id = models.CharField(max_length=30)
+    uid = models.CharField(max_length=30)
+    name = models.CharField(max_length=50)
+    type = models.CharField(max_length=20)
+
+
+class DistributionTreeRepository(Repository):
```

Comment:
Can this be inherited from `Model` and have it have an attribute that is a ForeignKey to repository.